users/src/main/java/io/crate/operation/auth/UserServiceFactoryImpl.java

```diff
@@ -22,13 +22,41 @@
 
 package io.crate.operation.auth;
 
+import io.crate.operation.collect.sources.SysTableRegistry;
+import io.crate.operation.user.*;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import java.util.concurrent.CompletableFuture;
 
 
 public class UserServiceFactoryImpl implements UserServiceFactory {
 
     @Override
-    public Authentication authService(Settings settings) {
-        return new HostBasedAuthentication(settings);
+    public Authentication authService(Settings settings, UserManager userManager) {
+        return new HostBasedAuthentication(settings, userManager);
+    }
+
+    @Override
+    public UserManager userManager(Settings settings,
```

Comment:
Maybe rename to `createUserManager` or `setupUserManager`- I'd also indicate with comments that this must only be called once as it registers transport actions